### PR TITLE
fix: nested bold breaks syntax (close #247)

### LINKED ISSUE #247: nested bold breaks syntax
## Version

1.2.3

## Test Environment

chrome

## Current Behavior

given text with a bold
```md
text1**text2** text3
```
select a whole line and click the bold toolbar button, see the rendered result

```md
**text1**text2** text3**
```
image

## Expected Behavior

we should consider the context of selected range or where the cursor is.
- the selected range overlaps an existing bold
- the selected range contains an existing bold

Taking care of overlap will involve several jobs.
we'll first un-bold the range which contains an existing bold(strike, italic, code)

### PR DESCRIPTION
**Pull request by kyuwoo-choi**
### Please check if the PR fulfills these requirements
- [x] It's the right issue type on the title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] It does not introduce a breaking change or has a description of the breaking change

### Description

#247 

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/js/markdownCommands/bold.js [hunk 1]
```diff
@@ -109,6 +117,15 @@ const Bold = CommandManager.command('markdown', /** @lends Bold */{
    */
   setCursorToCenter(doc, cursor) {
     doc.setCursor(cursor.line, cursor.ch + 2);
+  },
+
+  /**
+   * remove bold syntax in the middle of given text
+   * @param {string} text - text selected
+   * @returns {string} - un-bold text
```
Comment (seonim-ryu) [inline src/js/markdownCommands/bold.js]: `@private` 추가요~

File: src/js/markdownCommands/code.js [hunk 2]
```diff
@@ -4,6 +4,9 @@
 */
 import CommandManager from '../commandManager';
 
+const codeRangeRegex = /^`([^`]+)`$/;
+const codeContentRegex = /`([^`]+)`/g;
```
[the remark this answers is not in the document]
Comment (kyuwoo-choi) [reply src/js/markdownCommands/code.js]: 잠시 고민해봤습니다만, 공통으로 쓰일일이 전혀 없어서요.

File: src/js/markdownCommands/code.js [hunk 3]
```diff
@@ -21,25 +24,73 @@ const Code = CommandManager.command('markdown', /** @lends Code */{
...
+   * @returns {string} - text after code syntax removed
+   */
+  remove(text) {
+    return text.substr(1, text.length - 2);
+  },
+
+  /**
+   * remove bold syntax in the middle of given text
+   * @param {string} text - text selected
+   * @returns {string} - un-bold text
```
[the remark this answers is not in the document]
Comment (kyuwoo-choi) [reply src/js/markdownCommands/code.js]: 옙

File: src/js/markdownCommands/italic.js [hunk 4]
```diff
@@ -185,6 +193,15 @@ const Italic = CommandManager.command('markdown', /** @lends Italic */{
...
     doc.setCursor(cursor.line, cursor.ch + pos);
+  },
+
+  /**
+   * remove italic syntax in the middle of given text
+   * @param {string} text - text selected
+   * @returns {string} - text eliminated all italic in the middle of it's content
+   */
+  _removeItalicSyntax(text) {
+    return text ? text.replace(italicContentRegex, '$1') : '';
```
[the remark this answers is not in the document]
Comment (kyuwoo-choi) [reply src/js/markdownCommands/italic.js]: 이건 사실 한줄짜리라 메소드로 뺄 이유도 없지만,
코드 읽기 편하라고 의도 써주는 용도로만 쓰이고 있습니다.

공통처리를 하면 오히려 한번 더 움직여서 읽기 힘들어 질 것 같아요.

**Pull request merged**